tools/sof_ri_info/sof_ri_info.py

```diff
@@ -0,0 +1,883 @@
+                            0xbe, 0x77, 0xa2, 0x9c, 0x83, 0xf2, 0x25, 0x54,
+                            0x22, 0xfe, 0x29, 0x66, 0x5, 0xc2, 0xc9, 0x6b,
+                            0x8b, 0xa6, 0xa3, 0xf9, 0xb1, 0x6b, 0xaf, 0xe7,
+                            0x14, 0x77, 0xff, 0x17, 0xc9, 0x7c, 0x7c, 0x4e,
+                            0x83, 0x28, 0x2a, 0xe5, 0xc3, 0xcc, 0x6e, 0x25,
+                            0xa, 0x62, 0xbb, 0x97, 0x44, 0x86, 0x7c, 0xa2,
+                            0xd4, 0xf1, 0xd4, 0xf8, 0x8, 0x17, 0xf4, 0x6c,
+                            0xcc, 0x95, 0x99, 0xd4, 0x86, 0x37, 0x4, 0x9f,
+                            0x5, 0x76, 0x1b, 0x44, 0x55, 0x75, 0xd9, 0x32,
+                            0x35, 0xf1, 0xec, 0x4d, 0x93, 0x73, 0xe6, 0xc4])
+
+KNOWN_KEYS = {CNL_INTEL_PROD_KEY : 'CNL Intel prod key',
+              ICL_INTEL_PROD_KEY : 'ICL Intel prod key'}
+
+def parse_params():
+    """ Parses parameters
+    """
+    parser = argparse.ArgumentParser(description='SOF Binary Info Utility')
+    parser.add_argument('-v', '--verbose', help='increase output verbosity',
+                        action='store_true')
+    parser.add_argument('--headers', help='display headers only',
+                        action='store_true')
+    parser.add_argument('--full_bytes', help='display full byte arrays',
+                        action='store_true')
+    parser.add_argument('--no_colors', help='disable colors in output',
+                        action='store_true')
+    parser.add_argument('sof_ri_path', help='path to fw binary file to parse')
+    parsed_args = parser.parse_args()
+
+    return parsed_args
+
+# Helper Functions
+
+def change_color(color):
+    """ Prints escape code to change text color
+    """
+    color_code = {'red':91, 'green':92, 'yellow':93, 'blue':94,
+                  'magenta':95, 'cyan':96, 'white':98, 'none':0}
+    return '\033[{}m'.format(color_code[color])
+
+def uint_to_string(uint, both=False):
+    """ Prints uint in readable form
+    """
+    if both:
+        return hex(uint) + ' (' + repr(uint) + ')'
+    return hex(uint)
+
+def date_to_string(date):
+    """ Prints BCD date in readable form
+    """
+    return date[2:6]+'/'+date[6:8]+'/'+date[8:10]
+
+def chararr_to_string(chararr, max_len):
+    """ Prints array of characters (null terminated or till max_len)
+        in readable form
+    """
+    out = ''
+    for i in range(0, max_len):
+        if chararr[i] == 0:
+            return out
+        out += '{:c}'.format(chararr[i])
+    return out
+
+def mod_type_to_string(mod_type):
+    """ Prints module type in readable form
+    """
+    out = '('
+    # type
+    if (mod_type & 0xf) == 0:
+        out += ' builtin'
+    elif (mod_type & 0xf) == 1:
+        out += ' loadable'
+    #
+    if ((mod_type >> 4) & 0x1) == 1:
+        out += ' auto_start'
+    #
+    if ((mod_type >> 5) & 0x1) == 1:
+        out += ' LL'
+    if ((mod_type >> 6) & 0x1) == 1:
+        out += ' DP'
+    #
```

Comment:
Why are there empty comments?
What do LL/DP mod types mean?

Reply:
Completed the comments, no empty ones left.